pyfzz/fzz.cpp

```diff
@@ -0,0 +1,253 @@
+        for (const auto &it : id_maps.at(i)) { 
+            del_ids.push_back(it.second); 
+            orig_f_del_id.push_back(orig_f_id);
+            orig_f_id ++;
+        }
+    }
+
+    assert(del_ids.size() == s_id-1);
+    delete p_id_maps;
+
+    assert(simp_num == del_ids.size());
+
+    std::vector<Integer> cone_sid(simp_num+1);
+
+    for (auto del_id_it = del_ids.rbegin(); del_id_it != del_ids.rend(); ++del_id_it) {
+        bound_c.clear();
+        bound_c.push_back(*del_id_it);
+
+        std::vector<phat::index> orig_bound_c;
+        bound_chains.get_col(*del_id_it, orig_bound_c);
+
+        if (orig_bound_c.size() == 0) {
+            bound_c.push_back(0);
+        } else {
+            for (auto bsimp : orig_bound_c) {
+                // assert(cone_sid[bsimp] >= 0);
+                bound_c.push_back(cone_sid[bsimp]);
+            }
+        }
+
+        std::sort(bound_c.begin(), bound_c.end());
+
+        bound_chains.set_col(s_id, bound_c);
+        bound_chains.set_dim(s_id, getDim(bound_c));
+
+        cone_sid[*del_id_it] = s_id;
+
+        s_id ++;
+    }
+
+    phat::persistence_pairs pairs;
+    phat::compute_persistence_pairs< phat::twist_reduction >( pairs, bound_chains );
+
+    for (phat::index idx = 0; idx < pairs.get_num_pairs(); idx++) {
+            Integer b = pairs.get_pair(idx).first;
+            Integer d = pairs.get_pair(idx).second - 1;
+            Integer p = bound_chains.get_dim(b);
+
+            if (d < simp_num) { mapOrdIntv(b, d); } 
+            else { mapRelExtIntv(p, b, d); }
+
+            if (b > filt_simp.size()) { continue; }
+            if (d > filt_simp.size()) { d = filt_simp.size(); }
+            persistence->emplace_back(b, d, p);
+    }
+}
+std::vector<std::tuple<Integer, Integer, Integer>> FastZigzag::compute_zigzag(const std::vector<std::tuple<char, std::vector<int>>> &filt_simp) {
+    std::vector<Simplex> filt_simp_lin;
+    std::vector<bool> filt_op;
+    
+    // filt_simp_lin.emplace_back();
+    
+    for (auto i = 0; i < filt_simp.size(); i++) {
+        auto op = std::get<0>(filt_simp[i]);
+        auto simp = std::get<1>(filt_simp[i]);
+        // filt_simp_lin.emplace_back();
+       
+        FZZ::Simplex s;
+        if (op == 'i') {
+            filt_op.push_back(true);
+        } else {
+            assert(op == 'd');
+            filt_op.push_back(false);
+        }
+        
+        for(auto k = 0; k < simp.size(); k++){
+            s.push_back(simp[k]);
+            
+        }
+        // std::cout << std::endl;
+
+        filt_simp_lin.push_back(s);
+        
+    }
+    std::vector< std::tuple<FZZ::Integer, FZZ::Integer, FZZ::Integer> > persistence;
+    // FZZ::FastZigzag fzz;
+    compute(filt_simp_lin, filt_op, &persistence);
+    return persistence;
+}
+} // namespace FZZ {
+
+PYBIND11_MODULE(fzz, m) {
+    py::class_<FZZ::FastZigzag>(m, "fzz")
+        .def(py::init<>())
+        .def("compute_zigzag", &FZZ::FastZigzag::compute_zigzag);
+}
+
+
```

Comment:
Hi Soham, are you making a new copy the "fzz.h" and "fzz.cpp" files into a new folder 'pyfzz'? In that case, probably you shouldn't copy two new files and should reuse the existing ones instead.